packages/core/src/modules/x509/X509ModuleConfig.ts

```diff
@@ -0,0 +1,27 @@
+export interface X509ModuleConfigOptions {
+  trustedCertificates?: [string, ...string[]]
+}
+
+export class X509ModuleConfig {
+  private options: X509ModuleConfigOptions
+
+  public constructor(options?: X509ModuleConfigOptions) {
+    this.options = options ?? {}
+  }
+
+  public get trustedCertificates() {
+    return this.options.trustedCertificates
+  }
+
+  public setTrustedCertificates(trustedCertificates?: [string, ...string[]]) {
+    this.options.trustedCertificates = trustedCertificates ? [...trustedCertificates] : undefined
+  }
+
+  public addTrustedCertificate(trustedCertificate: string) {
+    if (!this.options.trustedCertificates) {
+      this.options.trustedCertificates = [trustedCertificate]
+      return
+    }
+    this.options.trustedCertificates.push(trustedCertificate)
+  }
```

Comment:
You are still overwriting the input options. This is not immutable. We should add an `#trustedCertificates` parameter to this class that is initlaized based on the [...options.trustedCertificates] . And when you call set/add you overwrite the / push to #trustedCertificates. This way the user input is not modified